test/utils/http.go

```diff
@@ -35,6 +35,68 @@ func GetRequest(url string) ([]byte, error) {
 	return request(req)
 }
 
+// GetRequestWithStatus sends a GET request and returns the response body, status code, and error.
+// Unlike GetRequest, this function does not treat non-200 status codes as errors,
+// allowing callers to explicitly check for specific status codes like 404.
+func GetRequestWithStatus(url string) ([]byte, int, error) {
```

Comment:
The old methods above are now unused. Remove them or combine them. The old methods used less redundant code. This could probably also be used in the new methods.